01_fib/fibonacci.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import "fmt"
+
+func fib(n int) {
+
+	var (
+		fn int
+		f0 int
+		f1 = 1
+	)
+
+	for i := 1; i <= n; i++ {
+		if i <= 1 {
+			fn = 1
+		} else {
+			fn = f0 + f1
+			f0 = f1
+			f1 = fn
```

Comment:
You can update in one line: `f0, f1 = f1, fn`. In fact, you could fold all three assignments into one: `fn, f0, f1 = f0 + f1, f1, fn`, but that's less clear, not more.

Reply:
good one @anzdaddy , did not know about the feature.

Reply:
No probs! I also just realised you don't need fn at all. I'll leave it to you to figure that one out. 🙂

Reply:
cleaned this up, thank you